BootloaderCommonPkg/Library/SecureBootLib/SecureBootRsa.c

```diff
@@ -61,16 +88,31 @@ DoRsaVerify (
     TmpPubKey[RSA_MOD_SIZE + Index] = InpPubKey->PubKeyData[RSA_MOD_SIZE + RSA_E_SIZE - 1 - Index];
   }
 
-  Status = DoHashVerify ((CONST UINT8 *)PubKeyBuf, RSA_MOD_SIZE + RSA_E_SIZE, HASH_TYPE_SHA256, ComponentType, PubKeyHash);
+  Status = DoHashVerify ((CONST UINT8 *)PubKeyBuf, RSA_MOD_SIZE + RSA_E_SIZE, SignHashAlg, ComponentType, PubKeyHash);
   if (RETURN_ERROR (Status)) {
     return Status;
   }
 
+
   // Verify payload data
-  Sha256 (Data, Length, Digest);
+  if (SignHashAlg == HASH_TYPE_SHA256) {
```

Comment:
I think you will need to use feature PCD for the code below.  Without it, you force all API to be linked in which increases the final binary size.

Reply:
Secureboot lib is designed to be generic to handle all cases for hashes.
For eg, hash alg in measured boot could be SM3 and hash used verified boot case can use SHA256. Also, I think there would be verified boot cases as "RSA3K/SHA384 and RSA2K/SHA256" need to be supported simultaneously based on info from IASImage/Container format.

Hash wrappers as sha256.c in IPPCryptoLib is protected with "PcdIppHashLibSupported". User could define the supported algorithms in this PCD.